### PR TITLE
Allow AbstractSerializationService.serializerFor() to be overridden

### PR DESCRIPTION
**Pull request by gierlachg**
Changed the visibility and allowed overriding of `AbstractSerializationService.serializerFor()` to be able to implement a `SerializationService` with dynamic serializer lookup.

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/ByteArraySerializerAdapter.java [hunk 1]
```diff
@@ -25,11 +25,11 @@
 
 import java.io.IOException;
 
-class ByteArraySerializerAdapter implements SerializerAdapter {
+public class ByteArraySerializerAdapter implements SerializerAdapter {
```
Comment (vbekiaris) [inline hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/ByteArraySerializerAdapter.java]: no need for this to be public as discussed

File: hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/AbstractSerializationService.java [hunk 2]
```diff
@@ -435,7 +446,7 @@ private SerializerAdapter registerFromSuperType(final Class type, final Class su
         return serializer;
     }
 
-    protected final SerializerAdapter serializerFor(final int typeId) {
+    public SerializerAdapter serializerFor(final int typeId) {
```
[the remark this answers is not in the document]
Comment (vbekiaris) [reply hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/AbstractSerializationService.java]: as discussed, `serializerFor` methods need to be `public` to be used by the delegating serialization service on Jet side.

File: hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/AbstractSerializationService.java [hunk 3]
```diff
@@ -445,7 +456,7 @@ protected final SerializerAdapter serializerFor(final int typeId) {
         return idMap.get(typeId);
     }
 
-    protected final SerializerAdapter serializerFor(Object object) {
+    public SerializerAdapter serializerFor(Object object) {
```
[the remark this answers is not in the document]
Comment (vbekiaris) [reply hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/AbstractSerializationService.java]: same as above `serializerFor` comment

**Comment by mmedenjak** [comment]
run-lab-run

**Pull request merged**